### PR TITLE
Add Meilisearch service

### PR DESCRIPTION
**Pull request by ankhers**
An open-source, lightning-fast, and hyper-relevant search engine that fits effortlessly into your workflow.

### CODE REVIEW

**Review by sandydoo — CHANGES_REQUESTED** [review]
Thanks! We could use an example that simply enables the service to test this in our CI.

**Comment by sandydoo** [comment]
`aarch64-linux` builds don't work right now. This'll be fixed in nixpkgs: https://github.com/NixOS/nixpkgs/pull/234258

File: src/modules/services/meilisearch.nix [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+      type = types.port;
+    };
+
+    environment = lib.mkOption {
+      description = lib.mdDoc "Defines the running environment of MeiliSearch.";
+      default = "development";
+      type = types.enum [ "development" "production" ];
+    };
+
+    # TODO change this to LoadCredentials once possible
```
[the remark this answers is not in the document]
Comment (ankhers) [reply src/modules/services/meilisearch.nix:47]: You are correct. I meant to remove it but forgot to after finishing.

**Comment by sandydoo** [comment]
The x86_64 macOS test will be fixed once https://github.com/NixOS/nixpkgs/pull/234678 lands on unstable.

File: src/modules/services/meilisearch.nix [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+in
+{
+  imports = [
+    (lib.mkRenamedOptionModule [ "meilisearch" "enable" ] [
+      "services"
+      "meilisearch"
+      "enable"
+    ])
+  ];
+
```
[the remark this answers is not in the document]
Comment (domenkozar) [reply src/modules/services/meilisearch.nix:16]: Nope.

[the remark this answers is not in the document]
Comment (ankhers) [reply src/modules/services/meilisearch.nix]: I just removed it.

**Pull request merged**